crates/kornia-io/src/png.rs

```
@@ -127,8 +178,13 @@ fn read_png_impl(file_path: impl AsRef<Path>) -> Result<(Vec<u8>, [usize; 2]), I
         return Err(IoError::InvalidFileExtension(file_path.to_path_buf()));
     }
 
-    let file = File::open(file_path)?;
-    let mut reader = Decoder::new(file)
+    let file = fs::read(file_path)?;
+    decode_png_imp(file.as_slice())
+}
+
+// Utility function to decode png files from raw bytes
+fn decode_png_imp(bytes: &[u8]) -> Result<(Vec<u8>, [usize; 2]), IoError> {
```

wondering if we should raise an error if expected type doesn't match

---

I was thinking that later, we could add a function like `decode_image` that will automatically going to figure out the file type, mode and that could internally use functions like `decode_image_png_rgba16`.

And for these functions, they could be used If the user know the file type and mode before hand or want the `Image` directly instead of a enum.